File: src/com/l2kt/gameserver/model/itemcontainer/PcWarehouse.java
package com.l2kt.gameserver.model.itemcontainer;

import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.model.item.instance.ItemInstance;

public class PcWarehouse extends ItemContainer
{
	private final Player _owner;
	
	public PcWarehouse(Player owner)
	{
		_owner = owner;
	}
	
	@Override
	public String getName()
	{
		return "Warehouse";
	}
	
	@Override
	public Player getOwner()
	{
		return _owner;
	}
	
	@Override
	public ItemInstance.ItemLocation getBaseLocation()
	{
		return ItemInstance.ItemLocation.WAREHOUSE;
	}
	
	@Override
	public boolean validateCapacity(int slots)
	{
		return _items.size() + slots <= _owner.getWareHouseLimit();
	}
}

File: src/com/l2kt/gameserver/model/itemcontainer/PcWarehouse.kt
package com.l2kt.gameserver.model.itemcontainer

import com.l2kt.gameserver.model.actor.instance.Player
import com.l2kt.gameserver.model.item.instance.ItemInstance

class PcWarehouse(public override val owner: Player) : ItemContainer() {

    override val name: String
        get() = "Warehouse"

    public override val baseLocation: ItemInstance.ItemLocation
        get() = ItemInstance.ItemLocation.WAREHOUSE

    override fun validateCapacity(slots: Int): Boolean {
        return _items.size + slots <= owner.wareHouseLimit
    }
}